locust/test/test_web.py

```diff
@@ -1083,3 +1083,63 @@ def test_request_stats_full_history_csv(self):
         self.assertEqual("/test2", rows[2][3])
         self.assertEqual("", rows[3][2])
         self.assertEqual("Aggregated", rows[3][3])
+
+
+class TestModernWebUi(LocustTestCase, _HeaderCheckMixin):
+    def setUp(self):
+        super().setUp()
+
+        parser = get_parser(default_config_files=[])
+        self.environment.parsed_options = parser.parse_args(["--modern-ui"])
+        self.stats = self.environment.stats
+
+        self.web_ui = self.environment.create_web_ui("127.0.0.1", 0, modern_ui=True)
+        self.web_ui.app.view_functions["request_stats"].clear_cache()
+        gevent.sleep(0.01)
+        self.web_port = self.web_ui.server.server_port
+
+    def tearDown(self):
+        super().tearDown()
+        self.web_ui.stop()
+        self.runner.quit()
+
+    def test_web_ui_reference_on_environment(self):
+        self.assertEqual(self.web_ui, self.environment.web_ui)
+
+    def test_web_ui_no_runner(self):
+        env = Environment()
+        web_ui = WebUI(env, "127.0.0.1", 0)
+        gevent.sleep(0.01)
+        try:
+            response = requests.get("http://127.0.0.1:%i/" % web_ui.server.server_port)
+            self.assertEqual(500, response.status_code)
+            self.assertEqual("Error: Locust Environment does not have any runner", response.text)
+        finally:
+            web_ui.stop()
+
+    def test_index_with_modern_ui(self):
+        self.assertEqual(200, requests.get("http://127.0.0.1:%i/" % self.web_port).status_code)
```

Comment:
Since every one of these tests relies on starting a new instance we can save a lot of test run overhead by just combining them into one.